### PR TITLE
[ScreenCaptureKit] Add ScreenCaptureKit bindings up to Xcode 13.3

### PR DESCRIPTION
**Pull request by dalexsoto**
Fixes https://github.com/xamarin/xamarin-macios/issues/14634

### CODE REVIEW

File: src/ScreenCaptureKit/SCContentFilter.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		{
+			switch (option) {
+			case SCContentFilterOption.Include:
+				Handle = InitWithDisplayIncludingWindows (display, windows);
+				break;
+			case SCContentFilterOption.Exclude:
+				Handle = InitWithDisplayExcludingWindows (display, windows);
+				break;
+			default:
+				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (option));
```
Comment (rolfbjarne) [inline src/ScreenCaptureKit/SCContentFilter.cs:50]: The value isn't null, so I think AOORE is more appropriate:

```suggestion
				ObjCRuntime.ThrowHelper.ThrowArgumentOutOfRangeException (nameof (option));
```

File: src/ScreenCaptureKit/SCContentFilter.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		{
+			switch (option) {
+			case SCContentFilterOption.Include:
+				Handle = InitWithDisplayIncludingApplications (display, applications, exceptingWindows);
+				break;
+			case SCContentFilterOption.Exclude:
+				Handle = InitWithDisplayExcludingApplications (display, applications, exceptingWindows);
+				break;
+			default:
+				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (option));
```
Comment (rolfbjarne) [inline src/ScreenCaptureKit/SCContentFilter.cs:65]: Same:

```suggestion
				ObjCRuntime.ThrowHelper.ThrowArgumentOutOfRangeException (nameof (option));
```

**Review by rolfbjarne — APPROVED** [review]
Looks good, just a couple of minor issues that can be fixed in a later PR if this needs to happen fast.

**Comment by rolfbjarne** [comment]
It turns out this is more complicated for Mac Catalyst, so I'm removing those bindings to get the macOS included.

The problem is that passing `-[weak_]framework ScreenCaptureKit` to the native linker for Mac Catalyst apps results in an error:

> ld: building for Mac Catalyst, but linking in .tbd built for macOS, file '/Applications/Xcode_13.3.0.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks//ScreenCaptureKit.framework/ScreenCaptureKit.tbd' for architecture x86_64

Now how does this work in Xcode?

1. The documentation says the ScreenCaptureKit is available on Mac Catalyst, but that none of the types in the framework are.
2. If you try to add ScreenCaptureKit to the "Link Binary With Library" setting in Xcode, ScreenCaptureKit doesn't show up.
3. If you try to add `@import ScreenCaptureKit;` to an Objective-C source file, that works fine.
4. If you try to use a class from ScreenCaptureKit, such as doing `NSLog (@"%@", [SCShareableContent class]);`, that works fine too.
5. Looking into the exact command line Xcode uses to link, it turns out that adding `-framework ScreenCaptureKit` to it causes the same linker error as we get.
6. Looking into the object file that contains the call to NSLog, clang added a `LC_LINKER_OPTION` section with the value `-framework ScreenCaptureKit`. This does not cause the linker to complain
7. Asking for verbose output from the linker, it tells us that it's loading the exact same `ScreenCaptureKit.tbd` file as it's complaining about when we pass `-framework ScreenCaptureKit` directly to the linker.

So it seems the linker is happy linking with ScreenCaptureKit when the `-framework ScreenCaptureKit` argument comes from a `LC_LINKER_OPTION` section in an object file, but not when passed on the command line.

**Comment by drasticactions** [comment]
@dalexsoto @rolfbjarne 

We don't need the bindings for Mac Catalyst (We're going to use it from a CLI macOS app) so I'm cool leaving it out for now.

File: src/ScreenCaptureKit/SCContentFilter.cs (hunk 2 above)
Comment (dalexsoto) [reply src/ScreenCaptureKit/SCContentFilter.cs]: Oops c/p mistake, good catch

**Comment by dalexsoto** [comment]
Thanks @rolfbjarne!

File: src/coregraphics.cs [hunk 3]
```diff
@@ -403,4 +403,48 @@ enum CGConstantColor {
...
+
+		[Field ("kCGDisplayStreamShowCursor")]
+		NSString ShowCursor { get; }
+
+		[Field ("kCGDisplayStreamSourceRect")]
+		NSString SourceRect { get; }
+
+		[Field ("kCGDisplayStreamYCbCrMatrix")]
+		NSString YCbCrMatrix { get; }
+	}
```
Comment (mandel-macaque) [inline src/coregraphics.cs:435]: Smart dict?

File: src/screencapturekit.cs [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+
+		[Field ("SCStreamFrameInfoContentScale")]
+		NSString ContentScale { get; }
+
+		[Field ("SCStreamFrameInfoContentRect")]
+		NSString ContentRect { get; }
+
+		[Field ("SCStreamFrameInfoDirtyRects")]
+		NSString DirtyRects { get; }
+	}
```
Comment (mandel-macaque) [inline src/screencapturekit.cs:84]: same.

**Review by mandel-macaque — APPROVED** [review]
We should look if we can add the strong dictionaries and land them.

**Comment by drasticactions** [comment]
FYI: Trying out the PR build, and I was able to access the APIs from macOS 12.3 just fine, including getting the windows and the required streams.

**Comment by dalexsoto** [comment]
Test failures are unrelated
* https://github.com/xamarin/maccore/issues/2560
* https://github.com/xamarin/maccore/issues/2558

**Pull request merged**